plenum/server/consensus/ordering_service.py

```diff
@@ -499,15 +503,19 @@ def process_preprepare(self, pre_prepare: PrePrepare, sender: str):
         """
         pp_key = (pre_prepare.viewNo, pre_prepare.ppSeqNo)
         # the same PrePrepare might come here multiple times
-        if (pp_key and (pre_prepare, sender) not in self.pre_prepare_tss[pp_key]):
+        if (pp_key and (pre_prepare.digest, sender) not in self.pre_prepare_tss[pp_key]):
```

Comment:
Why do we use `digest` here now?

Reply:
because storing PrePrepare message as part of key for inner dictonary looks strange and also requires, that PrePrepare should be hashable...